src/applications/disability-benefits/2346/components/SelectArrayItemsAccessoriesWidget.jsx

```diff
@@ -7,80 +10,140 @@ import {
   HEARING_AID_ACCESSORIES,
   WHITE_BACKGROUND,
 } from '../constants';
-// TODO: Safety checks for `selected` callback and `label` element
 
 class SelectArrayItemsAccessoriesWidget extends Component {
-  state = {
-    selectedItems: [],
-  };
-
-  handleChecked = e => {
-    e.persist();
-    if (e.target.checked) {
-      this.setState(prevState => ({
-        selectedItems: prevState.selectedItems.concat(e.target.name),
-      }));
+  handleChecked = (checked, supply) => {
+    const { selectedProducts, formData } = this.props;
+    let updatedSelectedProducts;
+    if (checked) {
+      updatedSelectedProducts = [
+        ...selectedProducts,
+        { productId: supply.productId },
+      ];
     } else {
-      this.setState(prevState => ({
-        selectedItems: prevState.selectedItems.filter(i => i !== e.target.name),
-      }));
+      updatedSelectedProducts = selectedProducts.filter(
+        selectedProduct => selectedProduct.productId !== supply.productId,
+      );
     }
+    const updatedFormData = {
+      ...formData,
+      selectedProducts: updatedSelectedProducts,
+    };
+    return this.props.setData(updatedFormData);
   };
   render() {
-    const { supplies } = this.props;
-    const { selectedItems } = this.state;
+    const { supplies, selectedProducts } = this.props;
+    const currentDate = moment();
 
-    return supplies.map(
-      supply =>
-        supply.productGroup === HEARING_AID_ACCESSORIES &&
-        supply.availableForReorder === true ? (
-          <div key={supply.productId} className="order-background">
-            <p className="vads-u-font-size--md vads-u-font-weight--bold">
-              {supply.productName}
-            </p>
-            <div className="vads-u-border-left--10px vads-u-border-color--primary-alt">
-              <div className="usa-alert-body vads-u-padding-left--1">
-                <p className="vads-u-margin--1px">
-                  <span className="vads-u-font-weight--bold">Battery: </span>
-                  {supply.productId}
-                </p>
-                <p className="vads-u-margin--1px">
-                  <span className="vads-u-font-weight--bold">Quantity: </span>
-                  {supply.quantity} <br />
-                  Approximately {supply.quantity} months supply
-                </p>
-                <p className="vads-u-margin--1px">
-                  <span className="vads-u-font-weight--bold">
-                    Last order date:{' '}
-                  </span>{' '}
-                  {moment(supply.lastOrderDate).format('MM/DD/YYYY')}
-                </p>
+    return (
+      <>
+        <h4>Select the hearing aid accessories you need</h4>
+        <p>
+          You'll be sent a 6-month supply for each accessory you choose below.
+        </p>
+        {supplies.map(
+          supply =>
+            supply.productGroup === HEARING_AID_ACCESSORIES &&
+            supply.availableForReorder === true ? (
```

Comment:
I think we should try to move away from all of the ternaries used here.  Before we do a map, we could do a filter and that way our array of products only includes the ones that we will show - from there we can map over those and output the JSX for those.  Something like:

```supplies.filter(supply => supply.productGroup === HEARING_AID_ACCESSORIES && supply.availableForReorder).map(supply => <jsx here>```

Reply:
Thanks, just pushed an update for this.